WordPress/src/main/java/org/wordpress/android/ui/domains/DomainRegistrationDetailsFragment.kt

```diff
@@ -320,53 +366,24 @@ class DomainRegistrationDetailsFragment : Fragment() {
                 address_second_line_input.text.toString(),
                 postal_code_input.text.toString(),
                 city_input.text.toString(),
-                selectedState?.code,
-                selectedCountry?.code,
+                null, // state code will be added in ViewModel
+                null, // country code will be added in ViewModel
                 email_input.text.toString(),
                 combinedPhoneNumber,
                 null
         )
     }
 
-    private fun showStatePicker(states: List<SupportedStateResponse>?) {
-        if (states == null || states.isEmpty()) {
-            return
-        }
-
-        val builder = AlertDialog.Builder(requireContext())
-        builder.setTitle(R.string.domain_registration_state_picker_dialog_title)
-        builder.setItems(states.map { it.name }.toTypedArray()) { _, which ->
-            selectedState = states[which]
-            state_input.setText(selectedState!!.name)
-        }
-
-        builder.setPositiveButton(R.string.dialog_button_cancel) { dialog, _ ->
-            dialog.dismiss()
-        }
-
-        builder.show()
+    private fun showStatePicker(states: List<SupportedStateResponse>) {
+        val dialogFragment = StatePickerDialogFragment.newInstance(states.toCollection(ArrayList()))
+        dialogFragment.setTargetFragment(this, 0)
+        dialogFragment.show(fragmentManager, StatePickerDialogFragment.TAG)
     }
 
     private fun showCountryPicker(countries: List<SupportedDomainCountry>) {
-        val builder = AlertDialog.Builder(requireContext())
-        builder.setTitle(R.string.domain_registration_country_picker_dialog_title)
-        builder.setItems(countries.map { it.name }.toTypedArray()) { _, which ->
-            val pickedCountry = countries[which]
-
-            if (selectedCountry != pickedCountry) {
-                selectedCountry = pickedCountry
-                country_input.setText(selectedCountry!!.name)
-                selectedState = null
-                state_input.text = null
-                fetchStates()
-            }
-        }
-
-        builder.setPositiveButton(R.string.dialog_button_cancel) { dialog, _ ->
-            dialog.dismiss()
-        }
-
-        builder.show()
+        val dialogFragment = CountryPickerDialogFragment.newInstance(countries.toCollection(ArrayList()))
+        dialogFragment.setTargetFragment(this, 0)
+        dialogFragment.show(fragmentManager, CountryPickerDialogFragment.TAG)
     }
 
     private fun showFormProgressIndicator() {
```

Comment:
NIT: this is a detail but I think it would be fine to have only one method with a boolean parameter that would set the view to visible/gone.